pkg/solana/transmitter.go

```diff
@@ -83,8 +83,10 @@ func (c *ContractTracker) Transmit(
 
 	// Send transaction, and wait for confirmation:
 	go func() {
+		ctx, cancel := context.WithTimeout(context.Background(), c.client.txTimeout)
+		defer cancel()
 		txSig, err := c.client.rpc.SendTransactionWithOpts(
-			context.Background(), // does not use libocr transmit context
+			ctx, // does not use libocr transmit context
```

Comment:
Add a 1m default context cancel to prevent leaky goroutines

Reply:
oh one potential issue I just noticed - is it possible SendTransactionWithOpts when it errors will return a nil txSig? If so that Debugf log will panic on txSig.String(). Probably want to early return on the error case

Reply:
hmmm it returns this: https://github.com/gagliardetto/solana-go/blob/d741961be7bd2b53d2a46e93518a6d573bb61f16/nativetypes.go#L82

but i'm not sure if it will be nil...i'll have to do some more digging

Reply:
nah you're safe then, arrays can't be nil. If it was a slice then it could be an issue